sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/serialization/BasicDigitalTwin.java

```diff
@@ -0,0 +1,109 @@
+
+
+package com.azure.digitaltwins.core.serialization;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.fasterxml.jackson.annotation.JsonInclude.*;
+
+/**
+ * An optional, helper class for deserializing a digital twin.
+ * Only properties with non-null values are included.
+ */
+@Fluent
+@JsonInclude(Include.NON_NULL)
+public class BasicDigitalTwin {
+
+    @JsonProperty(value = "$dtId", required = true)
+    private String id;
+
+    @JsonProperty(value = "$etag", required = true)
+    private String etag;
+
+    @JsonProperty(value = "$metadata", required = true)
+    private DigitalTwinMetadata metadata;
+
+    private final Map<String, Object> customProperties = new HashMap<>();
```

Comment:
Does this trick work in Java as well?

Reply:
Yeah, the `@JsonAnySetter` and `@JsonAnyGetter` annotations indicate that additional properties not mapped elsewhere should be put into this map.